source/recovery.rst

```diff
@@ -42,3 +42,122 @@ Applying this option forces the cluster to start. However, there may exist data
+
+3. Wait until the Pod ``0`` is ready, and execute the following code (it is
+   required for the Pod liveness check):
+
+   .. code-block:: bash
+
+      $ for i in $(seq 0 $(($(kubectl get pxc cluster1 -o jsonpath='{.spec.pxc.size}')-1))); do until [[ $(kubectl get pod cluster1-pxc-$i -o jsonpath='{.status.phase}') == 'Running' ]]; do sleep 10; done; kubectl exec cluster1-pxc-$i -- touch /var/lib/mysql/sst_in_progress; done
+
+4. Wait for all PXC Pods to start, then find the PXC instance with the most
+   recent data - i.e. the one with the highest `sequence number (seqno) <https://www.percona.com/blog/2017/12/14/sequence-numbers-seqno-percona-xtradb-cluster/>`_:
+
+   .. code-block:: bash
+
+      $ for i in $(seq 0 $(($(kubectl get pxc cluster1 -o jsonpath='{.spec.pxc.size}')-1))); do echo "###############cluster1-pxc-$i##############"; kubectl exec cluster1-pxc-$i -- cat /var/lib/mysql/grastate.dat; done
+
+   The output of this command should be similar to the following one::
+
+      ###############cluster1-pxc-0##############
+      # GALERA saved state
+      version: 2.1
+      uuid:    7e037079-6517-11ea-a558-8e77af893c93
+      seqno:   18
+      safe_to_bootstrap: 0
+      ###############cluster1-pxc-1##############
+      # GALERA saved state
+      version: 2.1
+      uuid:    7e037079-6517-11ea-a558-8e77af893c93
+      seqno:   18
+      safe_to_bootstrap: 0
+      ###############cluster1-pxc-2##############
+      # GALERA saved state
+      version: 2.1
+      uuid:    7e037079-6517-11ea-a558-8e77af893c93
+      seqno:   19
+      safe_to_bootstrap: 0
+
+   Now find the Pod with the largest ``seqno`` (it is ``cluster1-pxc-2`` in the
+   above example).
+
+5. Now execute the following commands *in a separate shell* to start this
+   instance:
+
+   .. code-block:: bash
+
+      $ kubectl exec cluster1-pxc-2 -- mysqld --wsrep_recover
+      $ kubectl exec cluster1-pxc-2 -- sed -i 's/safe_to_bootstrap: 0/safe_to_bootstrap: 1/g' /var/lib/mysql/grastate.dat
+      $ kubectl exec cluster1-pxc-2 -- sed -i 's/wsrep_cluster_address=.*/wsrep_cluster_address=gcomm:\/\//g' /etc/mysql/node.cnf
+      $ kubectl exec cluster1-pxc-2 -- mysqld
+
+   The ``mysqld`` process will initialize the database once again, and it will
+   be available for the incoming connections.
+
+6. Now go back to the original PXC image because the debug image is no longer
+   needed:
+
+   .. code-block:: bash
+
+      $ kubectl patch pxc cluster1 --type="merge" -p '{"spec":{"pxc":{"image":"perconalab/percona-xtradb-cluster-operator:master-pxc8.0"}}}'
+
+   .. note:: For PXC 5.7 this command should be as follows:
+
+      .. code-block:: bash
+
+         $ kubectl patch pxc cluster1 --type="merge" -p '{"spec":{"pxc":{"image":"perconalab/percona-xtradb-cluster-operator:master-pxc5.7"}}}'
+
+7. Go back *to the previous shell* and restart all Pods besides the
```

Comment:
I think we need to switch to prev shell in step 6.
because shell in step 5 become locked by mysqld